tools/ddrgen/src/scanners/dwarf/DwarfScanner.cpp

```diff
@@ -74,20 +74,25 @@ DwarfScanner::getBlacklist(Dwarf_Die die)
 
 	if (DW_DLV_ERROR == dwarf_hasattr(die, DW_AT_comp_dir, &hasAttr, &error)) {
 		ERRMSG("Checking for compilation directory attribute: %s\n", dwarf_errmsg(error));
-		goto Failed;
+		/* The DIE didn't have a compilationDirectory attribute so we can skip
```

Comment:
dwarf_hasattr() only returns DW_DLV_ERROR in the case of an internal dwarf library error, not in the case of no attribute present. If the Die has no compilation directory, hasAttr will be false. The correct thing to do here is to check the value of hasAttr, as was in the code somehow removed by the mentioned 055c3bf.